### DISCUSSION TITLE
Move all glyphs to start at U+F001 instead of U+F0001

### DISCUSSION THREAD

**Discussion by mhamid3d** [fontforge/fontforge#5127] [Q&A] [answered]
I'm trying to convert a UnicodeFull font to UnicodeBmp, but first I want to just move all the glyphs so they start at U+F001, the problem is if I just simply cut all the glyphs and paste, all their glyphs names are removed and set to the name of the unicode slot, I need them to retain their names.

**Comment by ctrlcctrlv** [comment, marked as answer]
This is untested but it ought to work.

```python3
import fontforge as ff
f = ff.activeFont()
for g in f.glyphs():
    if g.unicode <= 0xEFFF: continue
    g.unicode -= 0xE1000 # (0xF0001 - 0xF001)
```

**Reply by mhamid3d** [reply]
Incredible, thank you!
